src/libraries/Microsoft.PowerFx.Core/Texl/Builtins/Text.cs

```diff
@@ -99,33 +101,40 @@ public override bool CheckTypes(TexlNode[] args, DType[] argTypes, IErrorContain
                 return isValid;
             }
 
-            StrLitNode formatNode;
             if (!DType.String.Accepts(argTypes[1]))
             {
                 errors.EnsureError(DocumentErrorSeverity.Severe, args[1], TexlStrings.ErrStringExpected);
                 isValid = false;
             }
-            else if ((formatNode = args[1].AsStrLit()) != null)
+            else if (BinderUtils.TryGetConstantValue(checkTypesContext, args[1], out var formatArg))
             {
                 // Verify statically that the format string doesn't contain BOTH numeric and date/time
-                // format specifiers. If it does, that's an error accd to Excel and our spec.
-                var fmt = formatNode.Value;
+                // format specifiers. If it does, that's an error according to Excel and our spec.
 
                 // But firstly skip any locale-prefix
-                if (fmt.StartsWith("[$-"))
+                if (formatArg.StartsWith("[$-"))
                 {
-                    var end = fmt.IndexOf(']', 3);
+                    var end = formatArg.IndexOf(']', 3);
                     if (end > 0)
                     {
-                        fmt = fmt.Substring(end + 1);
+                        formatArg = formatArg.Substring(end + 1);
                     }
                 }
 
-                var hasDateTimeFmt = fmt.IndexOfAny(new char[] { 'm', 'd', 'y', 'h', 'H', 's', 'a', 'A', 'p', 'P' }) >= 0;
-                var hasNumericFmt = fmt.IndexOfAny(new char[] { '0', '#' }) >= 0;
+                var hasDateTimeFmt = formatArg.IndexOfAny(new char[] { 'm', 'd', 'y', 'h', 'H', 's', 'a', 'A', 'p', 'P' }) >= 0;
+                var hasNumericFmt = formatArg.IndexOfAny(new char[] { '0', '#' }) >= 0;
                 if (hasDateTimeFmt && hasNumericFmt)
                 {
-                    errors.EnsureError(DocumentErrorSeverity.Moderate, formatNode, TexlStrings.ErrIncorrectFormat_Func, Name);
+                    // Check if the date time format contains '0's after the seconds specifier, which
+                    // is used for fractional seconds - in which case it is valid
+                    var formatWithoutZeroSubseconds = Regex.Replace(formatArg, @"[sS]\.?(0+)",
```

Comment:
>Regex.Replace [](http://example.com/codeflow?start=54&length=13)

we always have to be wary of RegEx on the server. 
But I think this is ok since the regex is constant and avoids backtracing, etc (and not coming from user input)